### PR TITLE
webapp: misc bug fixes(create command, log download)

### LINKED ISSUE #3598: az webapp log download creates zip files that can't be opened on OSX and Linux
`az webapp log download` is meant to retrieve the web app logs for local analysis. If you use the command on OSX or *nix, the zip file that's downloaded can't be opened because it's in an invalid format. 

### Environment summary
Verified this behavior on OSX 10.12 and Ubuntu Trusty.

### PR DESCRIPTION
**Pull request by yugangw-msft**
1. Fix #3598. 
2. Fix a recent regression that creating node js web would fail for redundant app-setting `WEBSITE_NODE_DEFAULT_VERSION`
3. code clean ups

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py [hunk 1]
```diff
@@ -155,6 +152,50 @@ def test_create_in_different_group(self, resource_group, resource_group_location
...
+        webapp_name = self.create_random_name(prefix='webapp-win-log', length=24)
+        plan = self.create_random_name(prefix='win-log', length=24)
+        self.cmd('appservice plan create -g {} -n {}'.format(resource_group, plan))
+        self.cmd('webapp create -g {} -n {} --plan {} --deployment-source-url https://github.com/yugangw-msft/azure-site-test.git -r "node|6.1"'.format(resource_group, webapp_name, plan))
+        time.sleep(30)  # 30 seconds should be enough for the deployment finished(Skipped under playback mode)
+
+        # sanity check the traces
+        _, log_file = tempfile.mkstemp()
+        log_dir = log_file + '-dir'
+        log_file = log_file.replace('\\', '\\\\')
```
Comment (troydai) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py]: You don't have to do that. You just need to wrap the path with the double quote in the command one 171.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py [hunk 2]
```diff
@@ -155,6 +152,50 @@ def test_create_in_different_group(self, resource_group, resource_group_location
...
+        self.cmd('webapp log download -g {} -n {} --log-file {}'.format(resource_group, webapp_name, log_file))
+        zip_ref = zipfile.ZipFile(log_file, 'r')
+        zip_ref.extractall(log_dir)
+        p = Path(log_dir, r'LogFiles\kudu\trace')
+        self.assertTrue(p.is_dir())
```
Comment (troydai) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py]: Can this be done by `os.path.isdir(os.path.join(log_dir, 'LogFiles', 'kudu', 'trace'))`?

It's preferred to:
1. Avoid platform-specific separator (`\` vs `/`)
2. Use `os.path`

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py [hunk 3]
```diff
@@ -155,6 +152,50 @@ def test_create_in_different_group(self, resource_group, resource_group_location
...
+# Test Framework is not able to handle binary file format, hence, only run live
+class AppServiceLogTest(LiveScenarioTest):
+    @ResourceGroupPreparer()
+    def test_download_win_web_log(self, resource_group):
+        import zipfile
+        from pathlib import Path
+        webapp_name = self.create_random_name(prefix='webapp-win-log', length=24)
+        plan = self.create_random_name(prefix='win-log', length=24)
+        self.cmd('appservice plan create -g {} -n {}'.format(resource_group, plan))
+        self.cmd('webapp create -g {} -n {} --plan {} --deployment-source-url https://github.com/yugangw-msft/azure-site-test.git -r "node|6.1"'.format(resource_group, webapp_name, plan))
```
Comment (troydai) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py]: Is there another standardized source of sample project?

Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py]: So far, no. I can move the sample test project to an azure official repository, but figured better not
1. The sample is very simple and pretty much was copied and pasted from Node Express web sample. If lost, which hasn't happened, it is fairly trivial to recover.
2. It is tailor'd for CLI specific test scenarios, say outputting the node/python stack version, which is not very suitable to be used for general samples.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py (hunk 1 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py]: Thanks. This helps

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py (hunk 2 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py]: Thanks. I was a bit sloppy on the test code.)

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py (hunk 3 above)
Comment (troydai) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/test_webapp_commands.py]: Ok. Filed issue https://github.com/Azure/azure-cli/issues/4680 to tracking.

**Pull request merged**